sympy/core/add.py

```diff
@@ -486,9 +486,24 @@ def _eval_is_rational_function(self, syms):
     def _eval_is_algebraic_expr(self, syms):
         return all(term._eval_is_algebraic_expr(syms) for term in self.args)
 
+    def _eval_is_real(self):
+        def _n2(rv):
+            """Numeric testing for significant 2 digits for numbers"""
+            if rv is not None:
+                return rv
+            else:
+                if self.is_Number:
+                    try:
+                        return self.evalf(2, strict=True).is_real
+                    except PrecisionExhausted:
+                        return None
+                else:
+                    return None
+
+        return _n2(_fuzzy_group(
```

Comment:
I think you want something like return `fuzzy_or([_n2(self), _fuzzy_group(...)])` and 
```python
def _n2(self):
    if self.is_number:
        try:
            return self.evalf(2, strict=True).is_real
        except Precisionexhausted:
            pass
```

Reply:
Although the following fixes the issue (so EmptySet is returned instead of the error), it raises other issues (e.g. let q be the list that you give in the OP, then `re(q[0][0])` raises a recursion error). 
```diff
diff --git a/sympy/core/add.py b/sympy/core/add.py
index 9bb42c6..e48508b 100644
--- a/sympy/core/add.py
+++ b/sympy/core/add.py
@@ -487,7 +487,16 @@ def _eval_is_algebraic_expr(self, syms):
         return all(term._eval_is_algebraic_expr(syms) for term in self.args)
 
     # assumption methods
-    _eval_is_real = lambda self: _fuzzy_group(
+    def _eval_is_real(self):
+        if self.is_number:
+            r, i = self.n(2).as_real_imag()
+            if not i:
+                return True
+            if i._prec != 1:
+                if i != 0:
+                    return False
+                return True
+        return _fuzzy_group(
         (a.is_real for a in self.args), quick_exit=True)
     _eval_is_complex = lambda self: _fuzzy_group(
         (a.is_complex for a in self.args), quick_exit=True)
diff --git a/sympy/core/mul.py b/sympy/core/mul.py
index b2c760b..c5290c1 100644
--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -1178,6 +1178,16 @@ def _eval_is_real(self):
         return self._eval_real_imag(True)
 
     def _eval_real_imag(self, real):
+        if self.is_number:
+            r, i = self.n(2).as_real_imag()
+            if not real:
+                r, i = i, r
+            if not i:
+                return True
+            if i._prec != 1:
+                if i != 0:
+                    return False
+                return True
         zero = False
         t_not_re_im = None
 
diff --git a/sympy/core/power.py b/sympy/core/power.py
index 786368f..8f0a957 100644
--- a/sympy/core/power.py
+++ b/sympy/core/power.py
@@ -506,6 +506,14 @@ def _eval_is_integer(self):
 
     def _eval_is_real(self):
         from sympy import arg, exp, log, Mul
+        if self.is_number:
+            r, i = self.n(2).as_real_imag()
+            if not i:
+                return True
+            if i._prec != 1:
+                if i != 0:
+                    return False
+                return True
         real_b = self.base.is_real
         if real_b is None:
             if self.base.func == exp and self.base.args[0].is_imaginary:
```

I use `self.n(2).as_real_imag()` instead of `self.as_real_imag()` (with an evaluation of i later) because otherwise the i.n(2) ends up having a small imaginary part. 

(I would not consider this to be a challenging issue to fix.)